### DISCUSSION TITLE
JSON5 in shared presets repo not discovered by default

### DISCUSSION THREAD

**Discussion by reitzig** [renovatebot/renovate#23786] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.34.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

I'm extending presets like so:

```json5
{
  "extends" :      [
    "local>some-project/renovate-config",
  ],
}
```

With `default.json` in `some-project/renovate-config`, this worked, as per the [documentation](https://docs.renovatebot.com/config-presets/#shareable-config-presets).

After renaming the file to `default.json5` -- which is necessary for adopting JSON5 lest the IDE yells at me -- it does not.
FWIW, I prefer JSON5 for Renovate config files because trailing commas make it easier to copy-paste snippets, and I like to use comments to document my choices (_especially_ in presets).

From the docs, it is not clear to me whether that's intentional. It says "only use `default.json` for your presets" but also "shareable config presets must use the JSON or JSON5 formats". To me, that combines to "use to `default.json` or `default.json5`".

ℹ️ From the log, it seems clear that Renovate does not _try_ to resolve `default.json5` in the preset repo, so I have not created a pair of reproduction repos.

If the intent is to support `default.json5`, it would have to be added at [renovatebot/renovate:lib/config/presets/util.ts#32](https://github.com/renovatebot/renovate/blob/14f4617c77ecadcbed454fd80009d947c4eb3822/lib/config/presets/util.ts#L32). I don't quite see how to cover that in `util.spec.ts`, and my TypeScript is at caveman-level, so I think me creating a PR would create more work than it avoids. 😅

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: migrateAndValidate() (repository=some-project/my-repo)
DEBUG: No config migration necessary (repository=some-project/my-repo)
DEBUG: GET https://bitbucket.some.org/./rest/api/1.0/projects/some-project/repos/renovate-config/browse/default.json?limit=20000 = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=77) (repository=some-project/my-repo)
DEBUG: Preset file default.json not found in some-project/renovate-config (repository=some-project/my-repo)
DEBUG: GET https://bitbucket.some.org/./rest/api/1.0/projects/some-project/repos/renovate-config/browse/renovate.json?limit=20000 = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=74) (repository=some-project/my-repo)
DEBUG: Preset file renovate.json not found in some-project/renovate-config (repository=some-project/my-repo)
DEBUG: Preset fetch error (repository=some-project/my-repo)
       "preset": "local>some-project/renovate-config",
       "err": {
         "message": "dep not found",
         "stack": "Error: dep not found\n    at fetchJSONFile (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/local/common.ts:23:11)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at fetchPreset (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/util.ts:42:21)\n    at getPreset (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/index.ts:226:20)\n    at fetchPreset (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/index.ts:360:12)\n    at Object.resolveConfigPresets (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/index.ts:292:31)\n    at mergeRenovateConfig (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/init/merge.ts:260:5)\n    at getRepoConfig (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/init/config.ts:12:12)\n    at initRepo (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/init/index.ts:44:12)\n    at Object.renovateRepository (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/index.ts:54:14)\n    at attributes.repository (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/renovate.ts:18:22"
       }
 INFO: Throwing preset error (repository=some-project/my-repo)
       "validationError": "Cannot find preset's package (local>some-project/renovate-config)"
 INFO: Repository has invalid config (repository=some-project/my-repo)
       "error": {
         "validationError": "Cannot find preset's package (local>some-project/renovate-config)",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at fetchPreset (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/index.ts:371:19)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at Object.resolveConfigPresets (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/config/presets/index.ts:292:31)\n    at mergeRenovateConfig (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/init/merge.ts:260:5)\n    at getRepoConfig (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/init/config.ts:12:12)\n    at initRepo (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/init/index.ts:44:12)\n    at Object.renovateRepository (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/repository/index.ts:54:14)\n    at attributes.repository (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /opt/containerbase/tools/renovate/36.34.0/node_modules/renovate/lib/renovate.ts:18:22"
       }
DEBUG: raiseConfigWarningIssue() (repository=some-project/my-repo)
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
This is intentional. For every filename we add support to it would require an extra API call

**Reply by reitzig** [reply]
Pity. 

Soooo we can add `default.json5` once the deprecated `renovate.json` is removed? 😬

I guess replacing 1-2 (proposed: 1-3) single-file fetches with 1 (platform-dependent) directory listing plus 0-1 single-file fetches is out of the question?

**Reply by rarkins** [reply]
I think it might be possible today if you specify the full filename path but I'm on mobile this week so can't test it myself

**Reply by reitzig** [reply]
From what I saw of the code, yeah, that should work. However, it would mean I'd either have to break N repos and tell the resp. devs what to change, and/or update those I have access myself by hand. I'm not eager to do either. 😅

**Reply by reitzig** [reply]
For reference: #23911 ➡️ #23910

**Reply by HenrikPoulsen** [reply]
Would have been nice if there was an environment variable to control this however. I am self-hosting Renovate on GHES and github.com and would have liked if this was possible to control.
Being able to comment these presets when they are using in thousands of repos using them is a big plus. Them not being discoverable in any way for onboarding PRs makes it a bit complicated.
I know is being a self-hosted only setting would suck for others. Though there's already a number of self-hosted only options so I don't think it would be something strange to have
